src/main.cpp

```diff
@@ -936,6 +937,16 @@ bool ContextualCheckTransaction(
         }
     }
 
+    // Rules that apply to Canopy or later:
+    if (CanopyActive) {
+        BOOST_FOREACH(const JSDescription& joinsplit, tx.vJoinSplit)
+        {
+            if (joinsplit.vpub_old) {
```

Comment:
We should explicitly do `if (joinsplit.vpub_old > 0)` and not rely on implicit casting.